3.10/aql/functions-document.md

```diff
@@ -345,142 +622,365 @@ individual parts as separate attributes.
+    @startDocuBlockInline aqlUnsetRecursive_3
+    @EXAMPLE_AQL{aqlUnsetRecursive_3}
+    LET doc = { foo: { bar: { foo: 1, baz: 2 }, baz: 3 }, baz: 4 }
+    RETURN UNSET_RECURSIVE(doc, "baz")
+    @END_EXAMPLE_AQL
+    @endDocuBlock aqlUnsetRecursive_3
+    {% endaqlexample %}
+    {% include aqlexample.html id=examplevar type=type query=query bind=bind result=result %}
+
+Recursively remove multiple attributes (`foo` and `bar`):
+
+    {% aqlexample examplevar="examplevar" type="type" query="query" bind="bind" result="result" %}
+    @startDocuBlockInline aqlUnsetRecursive_4
+    @EXAMPLE_AQL{aqlUnsetRecursive_4}
+    LET doc = { foo: { bar: { foo: 1, baz: 2 }, baz: 3 }, baz: 4 }
+    RETURN UNSET_RECURSIVE(doc, "foo", "bar")
+    @END_EXAMPLE_AQL
+    @endDocuBlock aqlUnsetRecursive_4
+    {% endaqlexample %}
+    {% include aqlexample.html id=examplevar type=type query=query bind=bind result=result %}
+
+Recursively remove multiple attributes (`foo` and `baz`), removing all
+attributes of the example object:
+
+    {% aqlexample examplevar="examplevar" type="type" query="query" bind="bind" result="result" %}
+    @startDocuBlockInline aqlUnsetRecursive_5
+    @EXAMPLE_AQL{aqlUnsetRecursive_5}
+    LET doc = { foo: { bar: { foo: 1, baz: 2 }, baz: 3 }, baz: 4 }
+    RETURN UNSET_RECURSIVE(doc, "foo", "baz")
+    @END_EXAMPLE_AQL
+    @endDocuBlock aqlUnsetRecursive_5
+    {% endaqlexample %}
+    {% include aqlexample.html id=examplevar type=type query=query bind=bind result=result %}
+
+Recursively remove multiple attributes (`foo`, `bar`, and `baz`), removing all
+attributes of the example object:
+
+    {% aqlexample examplevar="examplevar" type="type" query="query" bind="bind" result="result" %}
+    @startDocuBlockInline aqlUnsetRecursive_6
+    @EXAMPLE_AQL{aqlUnsetRecursive_6}
+    LET doc = { foo: { bar: { foo: 1, baz: 2 }, baz: 3 }, baz: 4 }
+    RETURN UNSET_RECURSIVE(doc, "foo", "bar", "baz")
+    @END_EXAMPLE_AQL
+    @endDocuBlock aqlUnsetRecursive_6
+    {% endaqlexample %}
+    {% include aqlexample.html id=examplevar type=type query=query bind=bind result=result %}
+
+---
 
 `UNSET_RECURSIVE(document, attributeNameArray) → doc`
 
 - **document** (object): a document / object
 - **attributeNameArray** (array): an array of attribute names as strings
-- returns **doc** (object): *document* without the specified attributes on
+- returns **doc** (object): *document* without the specified attributes at
   all levels (top-level as well as nested objects)
 
-```js
-UNSET_RECURSIVE( doc, [ "_id", "_key", "foo", "bar" ] )
-```
+**Examples**
+
+Recursively remove `baz` attributes, by passing an array with the attribute key:
+
+    {% aqlexample examplevar="examplevar" type="type" query="query" bind="bind" result="result" %}
+    @startDocuBlockInline aqlUnsetRecursive_7
+    @EXAMPLE_AQL{aqlUnsetRecursive_7}
+    LET doc = { foo: { bar: { foo: 1, baz: 2 }, baz: 3 }, baz: 4 }
+    RETURN UNSET_RECURSIVE(doc, ["baz"])
```

Comment:
Looks like a meaningless example comparing to original `UNSET_RECURSIVE( doc, [ "_id", "_key", "foo", "bar" ] )`.

Reply:
The KEEP_RECURSIVE and UNSET_RECURSIVE examples use the same example object and I don't really want to use an object that has nested, system attribute-like keys, which would be the only way to make it meaningful for the different query variations, while still using the original arguments for the function call here.

The point of the specific example is to show that you can pass an array. For the keep counterpart it made sense to show passing multiple keys but here it seemed better to let some attributes remain in the object.

I agree that many of the examples are contrived but making them more real-world would also make them less succinct.